File: docs/components_page/components/alert/dismiss.py
import dash_bootstrap_components as dbc
from dash import Input, Output, State, html

alert = html.Div(
    [
        dbc.Button(
            "Toggle alert with fade",
            id="alert-toggle-fade",
            className="me-1",
            n_clicks=0,
        ),
        dbc.Button(
            "Toggle alert without fade", id="alert-toggle-no-fade", n_clicks=0
        ),
        html.Hr(),
        dbc.Alert(
            "Hello! I am an alert",
            id="alert-fade",
            dismissable=True,
            is_open=True,
        ),
        dbc.Alert(
            "Hello! I am an alert that doesn't fade in or out",
            id="alert-no-fade",
            dismissable=True,
            fade=False,
            is_open=True,
        ),
    ]
)


@app.callback(
    Output("alert-fade", "is_open"),
    [Input("alert-toggle-fade", "n_clicks")],
    [State("alert-fade", "is_open")],
)
def toggle_alert(n, is_open):
    if n:
        return not is_open
    return is_open


@app.callback(
    Output("alert-no-fade", "is_open"),
    [Input("alert-toggle-no-fade", "n_clicks")],
    [State("alert-no-fade", "is_open")],
)
def toggle_alert_no_fade(n, is_open):
    if n:
        return not is_open
    return is_open

File: docs/components_page/components/alert/dismiss.R
library(dashBootstrapComponents)

alert <- div(
  list(
    dbcButton(
      "Toggle alert with fade",
      id = "alert-toggle-fade", n_clicks = 0,
      className = "me-1"
    ),
    dbcButton("Toggle alert without fade",
      id = "alert-toggle-no-fade",
      n_clicks = 0
    ),
    html$hr(),
    dbcAlert(
      "Hello! I am an alert",
      id = "alert-fade",
      dismissable = TRUE,
      is_open = TRUE
    ),
    dbcAlert(
      "Hello! I am an alert that doesn't fade in or out",
      id = "alert-no-fade",
      dismissable = TRUE,
      fade = FALSE,
      is_open = TRUE
    )
  )
)


app$callback(
  output("alert-fade", "is_open"),
  list(
    input("alert-toggle-fade", "n_clicks"),
    state("alert-fade", "is_open")
  ),
  function(n, is_open) {
    if (n > 0) {
      return(!is_open)
    }
    return(is_open)
  }
)


app$callback(
  output("alert-no-fade", "is_open"),
  list(
    input("alert-toggle-no-fade", "n_clicks"),
    state("alert-no-fade", "is_open")
  ),
  function(n, is_open) {
    if (n > 0) {
      return(!is_open)
    }
    return(is_open)
  }
)
